rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -1368,7 +1375,26 @@ public void addSignature(BtcECKey federatorPublicKey, List<byte[]> signatures, b
         processSigning(federatorPublicKey, signatures, rskTxHash, btcTx, federation);
     }
 
-    private void processSigning(BtcECKey federatorPublicKey, List<byte[]> signatures, byte[] rskTxHash, BtcTransaction btcTx, Federation federation) throws IOException {
+    private Optional<Federation> getFederationFromPublicKey(BtcECKey federatorPublicKey) {
+        Federation retiringFederation = getRetiringFederation();
+        Federation activeFederation = getActiveFederation();
+
+        if (activeFederation.hasBtcPublicKey(federatorPublicKey)) {
+            return Optional.of(activeFederation);
+        }
+        if (retiringFederation != null && retiringFederation.hasBtcPublicKey(federatorPublicKey)) {
+            return Optional.of(retiringFederation);
+        }
+
+        return Optional.empty();
+    }
```

Comment:
This method can be an util method. This way, we reduce the size of `BridgeSupport` class and get it less overloaded.

Reply:
i think this class deserves a whole refactor

Reply:
Yes, it does. But the only way possible so far is by small steps at once. Consider moving it to a fed utils class

Reply:
Now that we have a FederationUtils class it could probably go in there. Having said that, I think that's another story since it will involve adding tests for this particular method

Reply:
okidoki